ttnn/cpp/ttnn/tensor/types.hpp

```diff
@@ -584,12 +591,9 @@ struct MultiDeviceHostStorage {
         // preinitialize empty tensor handles and use/populate them in the worker threads.
 
         inline void insert_buffer_and_shape_for_device(Device* device, const DeviceBuffer buffer, const Shape shape) {
+            std::scoped_lock lock(buffer_mtx, shape_mtx);
```

Comment:
so why previous code didn't work:
1st thread locks buffer_mtx and changes buffers.
Then 2nd thread comes and locks mutex and changes buffers.
BUT in general is unspecified which one thread comes to the second lock first.